File: 0590/solution.py
class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children


class Solution:
    def postorder(self, root: "Node") -> list[int]:
        post = []
        stack = []
        if root:
            stack.append(root)
        while stack:
            node = stack.pop()
            post.append(node.val)
            for child in node.children:
                stack.append(child)
        return post[::-1]

File: 0590/solution.ts
class _Node {
  val: number;
  children: _Node[];
  constructor(val?: number) {
    this.val = val === undefined ? 0 : val;
    this.children = [];
  }
}

function postorder(root: _Node | null): number[] {
  const post: number[] = [];
  const stack: _Node[] = [];
  if (root) {
    stack.push(root);
  }
  while (stack.length) {
    const node = stack.pop()!;
    post.push(node.val);
    for (const child of node.children) {
      stack.push(child);
    }
  }
  post.reverse();
  return post;
}
